e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js

```diff
@@ -817,6 +840,27 @@ function enableEmbeddingForResource({ resource, id }) {
   });
 }
 
+function assertValidPublicLink({ resource, shouldHaveRemoveLink = true }) {
```

Comment:
Let me know if there's an easier way to do this. I've added the regex because it will help us make sure that we're not getting any `null` values in tests

Reply:
Fair enough. Even though this looks a bit ugly, it gives me confident that the public link gotta be in a certain format, which is better than just asserting it's not null or empty.

Reply:
I'd suggest making `shouldHaveRemoveLink` required, this will force you to pass `shouldHaveRemoveLink` in every assertion and make the assertions clearer.

It's much more clear because you don't need to assume there would be default assertions somewhere else.

e.g. between
```ts
assertValidPublicLink({ resource: "question" });
```

VS
```ts
assertValidPublicLink({
  resource,
  shouldHaveRemoveLink: true,
});
```

It isn't that apparent here, but when you have a bunch of optional flags, when reading the test you gotta always think about how do I interpret this tests, what optional flags do I need to take into account, which could make reading the tests a lot harder.